pkg/worker/builder.go

```diff
@@ -116,3 +133,16 @@ func filterStages(stages []wharfyml.Stage, nameFilter string) []wharfyml.Stage {
 	}
 	return result
 }
+
+func shouldNotRunStage(anyStageFailed bool, s wharfyml.Stage) bool {
+	if s.RunsIf == wharfyml.StageRunsIfFail && anyStageFailed {
+		return false
+	}
+	if (s.RunsIf == "" || s.RunsIf == wharfyml.StageRunsIfSuccess) && !anyStageFailed {
+		return false
+	}
+	if s.RunsIf == wharfyml.StageRunsIfAlways {
```

Comment:
I think it's easier to read when the simple conditions are first. Consider moving to the top

Reply:
Changed to switch statement, should read nicer now 👍🏻